### PR TITLE
samsung-j5nlte: add accelerometer

### PR DESCRIPTION
**Pull request by nergzd723**

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 1]
```diff
@@ -31,6 +31,31 @@
...
+
+			vdd-supply = <&pm8916_l17>;
+			vddio-supply = <&pm8916_l5>;
+
+			reg = <0x1d>;
+			interrupt-parent = <&msmgpio>;
+			interrupts = <115 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "INT1";
+			
+			pinctrl-0 = <&accel_int_default>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:55]: pinctrl-0 always needs to be paired with pinctrl-names

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 2]
```diff
@@ -31,6 +31,31 @@
 		compatible = "linux,extcon-usb-gpio";
 		vbus-gpio = <&msmgpio 105 GPIO_ACTIVE_HIGH>;
 	};
+
+	i2c-sensors {
+		compatible = "i2c-gpio";
+
+		sda-gpios = <&msmgpio 31 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&msmgpio 32 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		pinctrl-names = "default";
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:40]: pinctrl-names always needs to be paired with pinctrl-0

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 3]
```diff
@@ -218,4 +243,24 @@
 			bias-pull-down;
 		};
 	};
+	accel_int_default: accel_int_default {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:246]:
```suggestion
	accel_int_default: accel-int-default {
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 4]
```diff
@@ -218,4 +243,24 @@
...
+	};
+
+	sensors_i2c1: sensors_i2c1 {
+		function = "gpio";
+		pins = "gpio31";
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	sensors_i2c2: sensors_i2c2 {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:260]: You can combine this to one pinctrl,

```
pins = "gpio31", "gpio32";
```

**Review by Minecrell — CHANGES_REQUESTED** [review]

**Comment by Minecrell** [comment]
Also, please check sensor according to https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Testing_2

**Comment by nergzd723** [comment]
> Also, please check sensor according to https://wiki.postmarketos.org/wiki/MSM8916_Mainlining#Testing_2

it works:

```
    Accelerometer orientation changed: right-up
    Accelerometer orientation changed: normal
    Accelerometer orientation changed: right-up
    Accelerometer orientation changed: bottom-up
    Accelerometer orientation changed: right-up
    Accelerometer orientation changed: left-up
    Accelerometer orientation changed: normal

```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 5]
```diff
@@ -218,4 +244,17 @@
...
 		};
 	};
+	accel_int_default: accel-int-default {
+		function = "gpio";
+		pins = "gpio115";
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	sensors_i2c: sensors_i2c {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:254]:
```suggestion
	sensors_i2c_default: sensors-i2c-default {
```

**Review by Minecrell — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 6]
```diff
@@ -31,6 +31,32 @@
...
 	};
+
+	i2c-sensors {
+		compatible = "i2c-gpio";
+
+		sda-gpios = <&msmgpio 31 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+		scl-gpios = <&msmgpio 32 (GPIO_ACTIVE_HIGH|GPIO_OPEN_DRAIN)>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&accel_int_default>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:42]: this one should have the i2c pinctrl, while the accel pinctrl should be on the accel node, obviously :)

**Review by Minecrell — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts [hunk 7]
```diff
@@ -218,4 +246,17 @@
...
+		drive-strength = <2>;
+		bias-disable;
+	};
+
+	sensors_i2c_default: sensors-i2c-default {
+		function = "gpio";
+		pins = "gpio31", "gpio32";
+		drive-strength = <2>;
+		bias-disable;
+	};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5nlte.dts:261]: pinctrl should be sorted alphabetically ^-^

**Review by Minecrell — CHANGES_REQUESTED** [review]

**Comment by Minecrell** [comment]
I fixed the rest up for you, but please order pinctrl alphabetically next time :)

**Review by Minecrell — APPROVED** [review]

**Comment by nergzd723** [comment]
> I fixed the rest up for you, but please order pinctrl alphabetically next time :)

Thanks :)

**Pull request merged**